docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +343,217 @@ recursion was explicitly requested.
+
+To achieve higher verification performance, Dafny collects all the assertions of one method into one single conjecture (an _assertion batch_) that it sends to the verifier, which tries to prove it correct:
+
+* If the verifier says it is correct[^smt-encoding], it means that all the assertions hold.
+* If the verifier returns a counter-example, this counter-example is used to determine both the failing assertion and the failing path.
+  Dafny will then query again the verifier with the assumption that that particular assert is valid, so that it can retrieve other failing assertions happening afterwards.[^caveat-about-assertion-and-assumption]
+* If the verifier returns `unknown` or times out, or even preemptively for difficult assertions or to reduce the chance that the verifier will ‘be confused’ by the many assertions in a large batch, Dafny partitions the assertions up into smaller batches[^smaller-batches]. An extreme case is the use of the `/oneAssertAtATime` command-line option, which causes Dafny to make one batch for each assertion.
+
+[^smt-encoding]: The formula sent to the underlying SMT solver is the negation of the formula that the verifier wants to prove. Hence, if the SMT solver returns "unsat", it means that the SMT formula is always false, meaning the verifier's formula is always true. On the other side, if the SMT solver returns "sat", it means that the SMT formula can be made true with a special variable assignment, which means that the verifier's formula is false under that same variable assignment, meaning it's a counter-example for the verifier.
+
+[^caveat-about-assertion-and-assumption]: Caveat about assertion and assumption: One big difference between an "assertion transformed in an assumption" and the original "assertion" is that the original "assertion" can unroll functions twice, whereas the "assumed assertion" can unroll them only once. Hence, Dafny can still continue to analyze assertions after a failing assertion without automatically proving "false" (which would make all further assertions to prove automatically).
+
+[^smaller-batches]: To create a smaller batch, Dafny duplicates the list of assertions, and ensure every assertion is transformed into an assumption in exactly one list, so that the assertions of the two new lists form a partition of the original list. This results in "easier" formulas for the verifier because it has less to prove, but it takes more overhead because every verification instance have a common set of axioms and there is no knowledge sharing between instances because they run independently.
```

Comment:
I think this isn't quite right. Each assertion appears as an assertion in exactly one list, and every other assertion in that list is transformed into an assumption.

Reply:
What about the reformulation `Dafny duplicates the list of assertions, and transforms each assertion into an assumption into exactly one list, so that assertions are verified only in one list.`

Reply:
I think it's the opposite of that. "Dafny duplicates the list of assertions (resulting in N lists if there are N assertions), and leaves each assertion in its original form in exactly one list, transforming the rest of that list into assumptions, so that each assertion is verified only in one list."

Reply:
I don't think this is how it works. What you describe is exactly the behavior of `/vcsSplitOnEveryAssert`, but splitting is more general than that.